goose3/extractors/opengraph.py

```diff
@@ -28,12 +28,28 @@ class OpenGraphExtractor(BaseExtractor):
 
     def extract(self):
         opengraph_dict = {}
+        og_type = None
         node = self.article.doc
         metas = self.parser.getElementsByTag(node, 'meta')
+
+        # Open Graph type that is supported. In theory it is possible
+        # that a page has multiple types, but we will only consider
+        # the first one.
+        og_types = [
+            self.parser.getAttribute(meta, 'content')
+            for meta in metas
+            if self.parser.getAttribute(meta, 'property') == "og:type"
+        ]
```

Comment:
Note: This is now `O(2n)`. In theory we could add the following block to the loop below to make it `O(n)`

```python
if not og_type and attr == "og:type":
    og_type = value
```
This change would not support the sub-optimal case where the `og:type` attribute appeared after the first attribute for the given type. See the `article:section` attribute in the added test and the resulting output for an example.